### PR TITLE
feat: handover sap elder checks with membership knowledge

### PR DESCRIPTION
**Pull request by grumbach**
Squashed version of PR #1215  
https://github.com/maidsafe/safe_network/pull/1215

### CODE REVIEW

File: sn_node/src/node/core/messaging/handling/handover.rs [hunk 1]
```diff
@@ -82,24 +87,146 @@ impl Node {
...
+            error!("Missing membership instance when checking handover SAP candidates");
+            Err(Error::MissingMembershipInstance)
+        }
+    }
+
+    async fn get_sap_for_prefix(&self, prefix: Prefix) -> Result<SectionAuthorityProvider> {
+        match self.network_knowledge.prefix_map().get(&prefix) {
+            Some(sap) => Ok(sap),
+            None => Err(Error::FailedToGetPreviousSAP),
+        }
```
Comment (davidrusu) [inline sn_node/src/node/core/messaging/handling/handover.rs:103]:
```suggestion
        self.network_knowledge.prefix_map().get(&prefix).ok_or(Error::FailedToGetPreviousSAP)
```

**Review by davidrusu — APPROVED** [review]

**Review by davidrusu — APPROVED** [review]

**Comment by joshuef** [comment]
looks like bors failed with a CI timeout, but no strong reason for it that I could see (bit of a long build perhaps). May be the timeout is just a weeee bit too short. Reborsed and we can increase the timeout separately

**Pull request merged**